packages/block-library/src/navigation/edit.js

```diff
@@ -57,10 +55,10 @@ function Navigation( {
 	//
 	// HOOKS
 	//
-	/* eslint-disable @wordpress/no-unused-vars-before-return */
 	const ref = useRef();
 	const { selectBlock } = useDispatch( 'core/block-editor' );
 
+	/* eslint-disable @wordpress/no-unused-vars-before-return */
```

Comment:
Do we find ourselves having to disable this often in components? If it's not a useful rule then we should just remove it...

(Not relevant to this PR!)

Reply:
The value of the rule is explained here:

https://github.com/WordPress/gutenberg/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars-before-return.md

Based on specific instances, we can choose to improve it as well. For example, it should probably be exempting hooks as a special circumstances, due to constraints imposed on us externally via [Rules of Hooks](https://reactjs.org/docs/hooks-rules.html).

Reply:
> The value of the rule is explained here:

I should have also mentioned there was previously quite a few _offending_ cases that were cleaned up as part of the introduction of the rule, in case that's a more convincing method for interpreting the value.

See: #12827

Reply:
It might not have been necessary to disable the rule here anyways? It was removed in #20884. If I recall correctly, the rule makes exceptions for object destructuring.

To the general problem: 

> we can choose to improve it as well. For example, it should probably be exempting hooks as a special circumstances, due to constraints imposed on us externally via [Rules of Hooks](https://reactjs.org/docs/hooks-rules.html).

This already exists as of #16737:

https://github.com/WordPress/gutenberg/blob/1f00c74c193253c2706f2729f5470573345ed823/packages/eslint-plugin/configs/react.js#L10-L15

The problem here is that the hook is named as experimental, `__experimentalUseColors`. There are two ways we could accommodate for this:

1. We can adopt a convention to reference the slot by its non-experimental name in the implementation of the component, renaming via the import ([example](https://github.com/WordPress/gutenberg/blob/1f00c74c193253c2706f2729f5470573345ed823/packages/block-editor/src/components/block-inspector/index.js#L12))
2. We can expand the pattern above to include `__experimental` and `__unstable` naming conventions